dev/com.ibm.ws.transaction.cloud_fat.base/test-applications/transaction/src/servlets/Simple2PCCloudServlet.java

```
@@ -482,18 +484,37 @@ public void setupNonUniqueLeaseLog(HttpServletRequest request,
     public void dropServer2Tables(HttpServletRequest request,
                                   HttpServletResponse response) throws Exception {
 
-        try (Connection con = getConnection(dsTranLog)) {
-            con.setAutoCommit(false);
-            try (Statement stmt = con.createStatement()) {
-                String dropTableString = "DROP TABLE WAS_TRAN_LOGCLOUD0021";
-                System.out.println("dropServer2Tables: Drop table using: " + dropTableString);
-                int dropReturn = stmt.executeUpdate(dropTableString);
-                dropTableString = "DROP TABLE WAS_PARTNER_LOGCLOUD0021";
-                System.out.println("dropServer2Tables: Drop table using: " + dropTableString);
-                dropReturn = stmt.executeUpdate(dropTableString);
+        boolean tablesExist = true;
+        final String[] types = { "TABLE" };
+        final List<String> tables = Arrays.asList("WAS_TRAN_LOGCLOUD0021", "WAS_PARTNER_LOGCLOUD0021");
+
+        while (tablesExist) {
+            try (Connection con = getConnection(dsTranLog)) {
+                con.setAutoCommit(false);
+                for (String table : tables) {
+                    final String dropTableString = "DROP TABLE " + table;
+                    try (Statement stmt = con.createStatement()) {
+                        final int dropReturn = stmt.executeUpdate(dropTableString);
+                        System.out.println("dropServer2Tables: " + dropTableString + " returned " + dropReturn);
+                    } catch (SQLSyntaxErrorException e) {
+                        System.out.println("dropServer2Tables: " + dropTableString + " threw " + e);
```

Throwable.toString() (used in this System.out.println()) doesn't capture the exception's stack trace, only its short description - is that what you want?

Also, should the exception be swallowed, so con.commit() is called), or rethrown, to avoid doing so?

---

We don't need the exception stack here. SQLSyntaxErrorException is thrown from the Oracle driver when the table being dropped already didn't exist. The exception message tells you that.

We can go ahead with the commit here. The surrounding while loop will handle a retry if not enough tables were dropped yet.